### PR TITLE
Unlink survey from a draft engagement

### PR DESCRIPTION
**Pull request by jadmsaadaot**
*Issue #:*
https://github.com/bcgov/met-public/issues/358

*Description of changes:*

- Add API resource, service and DB query to unlink survey
- Add front-end service and logic to click remove survey and call endpoint

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/services/comment_service.py [hunk 1]
```diff
@@ -79,7 +80,10 @@ def extract_comments(cls, survey_submission: SubmissionSchema, survey: SurveySch
     def review_comment(cls, comment_id, status_id, external_user_id):
         """Review comment."""
         user = UserService.get_user_by_external_id(external_user_id)
-        if not status_id or status_id == 1 or not user:
+
+        valid_statuses = [status.id for status in CommentStatus.get_comment_statuses()]
+
+        if not status_id or status_id == 1 or status_id not in valid_statuses or not user:
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/comment_service.py:86]: please replace 1 with CommentStatus.Pending.Value that way we know whats the 1 for

**Review by saravanpa-aot — CHANGES_REQUESTED** [review]

File: met-api/src/met_api/services/comment_service.py (hunk 1 above)
Comment (jadmsaadaot) [reply met-api/src/met_api/services/comment_service.py:86]: of course

File: met-api/src/met_api/resources/survey.py [hunk 2]
```diff
@@ -158,3 +158,28 @@ def put(survey_id, engagement_id):
             return ActionResult.error(str(err))
         except ValueError as err:
             return ActionResult.error(str(err))
+
+
+@cors_preflight('PUT,OPTIONS')
+@API.route('/<survey_id>/unlink/engagement/<engagement_id>')
```
[the remark this answers is not in the document]
Comment (jadmsaadaot) [reply met-api/src/met_api/resources/survey.py:164]: Understood, just concerned about using DELETE when we are only unlinking the survey from the engagement not deleting it from the records

**Review by djnunez-aot — APPROVED** [review]
cool way of using action context for the modal

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**